app/src/main/java/org/mozilla/fenix/settings/studies/StudiesView.kt

```diff
@@ -0,0 +1,129 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.studies
+
+import android.content.Context
+import android.text.SpannableStringBuilder
+import android.text.method.LinkMovementMethod
+import android.text.style.ClickableSpan
+import android.text.style.URLSpan
+import android.view.View
+import android.widget.TextView
+import androidx.annotation.VisibleForTesting
+import androidx.appcompat.widget.SwitchCompat
+import androidx.core.text.HtmlCompat
+import androidx.core.text.getSpans
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import mozilla.components.service.nimbus.NimbusApi
+import mozilla.components.support.base.log.logger.Logger
+import org.mozilla.experiments.nimbus.internal.EnrolledExperiment
+import org.mozilla.fenix.R
+import org.mozilla.fenix.databinding.SettingsStudiesBinding
+import org.mozilla.fenix.settings.SupportUtils
+import org.mozilla.fenix.utils.Settings
+
+@Suppress("LongParameterList")
+class StudiesView(
+    private val scope: CoroutineScope,
+    private val context: Context,
+    private val binding: SettingsStudiesBinding,
+    private val interactor: StudiesInteractor,
+    private val settings: Settings,
+    private val experiments: NimbusApi,
+    private val isAttached: () -> Boolean
+) : StudiesAdapterDelegate {
+    private val logger = Logger("StudiesView")
+
+    @VisibleForTesting
+    internal lateinit var adapter: StudiesAdapter
+
+    @Suppress("TooGenericExceptionCaught")
+    fun bind() {
+        provideStudiesTitle().text = getSwitchTitle()
```

Comment:
Are the function calls to get Views used here so bind() can be tested?

Reply:
Yes, I was having some troubles substituting the values on tests, so I created those function that I can easily mock from tests :)